### ISSUE TITLE
Sometimes the hashtag selection box won't go away

### ISSUE THREAD

**Issue by afontenot** [mastodon/mastodon#20793]
### Steps to reproduce the problem

1. Start writing a new post. Type a hashtag like `#mastodon` fully and hit the space bar at the end
2. Start writing a new post. Type a hashtag like `#mastodon`. Click somewhere in the text box outside of the hashtag to move the cursor.

### Expected behaviour

The hashtag selection box disappears.

### Actual behaviour

The hashtag selection box remains visible (not 100% reproducible for the spacebar case).

### Detailed description

Video showing the problem:

https://user-images.githubusercontent.com/3870410/201969852-6c9475f0-84d0-4c78-b347-3fd2f7bb4f2e.mp4

As you can see, I fail to reproduce it the first time I try. After I reproduce the issue with the space bar not hiding the box, I try clicking around for a while and you can see that it fails.

This issue is actually more annoying that it sounds at first because the hashtag box hides the "post" button. If you give up on making it go away and click on the hashtag you want in the box, then the cursor gets reset back to the end of the hashtag, even if you were busy typing somewhere else, which is mildly infuriating. 🙂

### Specifications

Mastodon 4.0.2 (mastodon.social)
Firefox 105

**Comment by vmstan** [comment]
@afontenot does this behavior persist in newer versions of Mastodon?

**Comment by afontenot** [comment]
@vmstan Yes I just confirmed it still happens on v4.2.1+pr-27634-22407c7 (current on dot social). Actually it's even worse now. Now, clicking a hashtag in the selection box *doesn't do anything at all* if you've already pressed space as shown in the video.

On the other hand, you can click outside the post box and that gets rid of the hashtag selection box, and I don't think that worked before. So that makes the problem slightly less annoying, but it's definitely still present.
